src/library/Tabs/Tabs.js

```diff
@@ -0,0 +1,292 @@
+      height,
+      position,
+      ...restProps,
+      'aria-labelledby': undefined
+    };
+
+    const tabListProps = {
+      align,
+      'aria-label': label,
+      'aria-labelledby': this.props['aria-labelledby'],
+      height,
+      onIncrement: this.handleIncrement,
+      position,
+      role: 'tablist',
+      vertical: this.vertical
+    };
+
+    return (
+      <Root {...rootProps}>
+        <TabList {...tabListProps}>{tabItems}</TabList>
+        <TabPanel {...selectedTabPanelProps} />
+      </Root>
+    );
+  }
+
+  handleClick = (event: AnchorEvent) => {
+    event.preventDefault();
+    event.persist();
+    const { currentTarget: target } = event;
+    const selectedTabIndex = parseInt(target.getAttribute('data-index'));
+
+    if (this.isControlled('selectedTabIndex')) {
+      this.changeActions(selectedTabIndex, event);
+    } else {
+      this.setSelectedTabIndex(selectedTabIndex, event);
+    }
+  };
+
+  handleKeyDown = (event: SyntheticKeyboardEvent<HTMLAnchorElement>) => {
+    event.persist();
+    if (['ArrowLeft', 'ArrowRight'].indexOf(event.key) !== -1) {
+      event.preventDefault();
+      const rtl = this.props.theme.direction === 'rtl';
+      const flippedDirections = {
+        ArrowLeft: 'ArrowRight',
+        ArrowRight: 'ArrowLeft'
+      };
+      const key =
+        rtl && flippedDirections[event.key]
+          ? flippedDirections[event.key]
+          : event.key;
+      this.handleIncrement(key, event);
+    } else if (
+      this.vertical &&
+      ['ArrowDown', 'ArrowUp'].indexOf(event.key) !== -1
+    ) {
+      event.preventDefault();
+      this.handleIncrement(event.key, event);
+    }
+  };
+
+  handleIncrement = (direction: string, event: SyntheticEvent<*>) => {
+    const selectedTabIndex = this.getControllableValue('selectedTabIndex');
+    const nextIndex =
+      direction === 'ArrowRight' || direction === 'ArrowDown'
+        ? selectedTabIndex === this.lastIndex
+          ? 0
+          : this.getNonDisabledIndex(selectedTabIndex + 1)
+        : selectedTabIndex === 0
+          ? this.lastIndex
+          : this.getNonDisabledIndex(selectedTabIndex - 1, { decrease: true });
+    this.setSelectedTabIndex(nextIndex, event);
+  };
+
+  getNonDisabledIndex = (
+    index: number,
+    { decrease }: { decrease: ?boolean } = {}
+  ) =>
+    this.disabledTabIndexes.has(index)
+      ? this.getNonDisabledIndex(decrease ? index - 1 : index + 1, { decrease })
+      : index;
+
+  setSelectedTabIndex = (selectedTabIndex: number, event: AnchorEvent) => {
+    this.setState(
+      {
+        selectedTabIndex
+      },
+      () => {
+        if (canUseDOM) {
+          const nextTab =
+            this.root &&
+            this.root.querySelector(`[data-index="${selectedTabIndex}"]`);
+          nextTab && nextTab.focus();
+        }
+        this.changeActions(selectedTabIndex, event);
+      }
+    );
+  };
+
+  changeActions = (selectedTabIndex: number, event: AnchorEvent) => {
```

Comment:
The `onChange` callback should only be invoked when a different tab is selected.

```js
selectedTabIndex !== this.getControllableValue('selectedTabIndex'))
```